src/authz-smart/smartHandler.ts

```diff
@@ -8,31 +8,130 @@ import {
     AuthorizationRequest,
     AuthorizationBundleRequest,
     AllowedResourceTypesForOperationRequest,
+    SystemOperation,
+    TypeOperation,
 } from 'fhir-works-on-aws-interface';
-import { SMARTConfig } from './smartConfig';
+import axios from 'axios';
+import { LaunchType, ScopeType, SMARTConfig } from './smartConfig';
 
 // eslint-disable-next-line import/prefer-default-export
 export class SMARTHandler implements Authorization {
+    static readonly SCOPE_REGEX = /(patient|user|system|launch)\/(\w+|\*)\.?(\w+|\*)?|(^launch$)/;

```

Comment:
In SMART are these the only four possible scopes? Would it be possible for `scopeRule` to have more than these 4 types, and therefore cause this regex to become invalid?

Reply:
the first part of the scope currently can only have 4 (this may change as the spec changes). I think this regex is complete - but would love to know if you see an errror

Clinical scopes: http://www.hl7.org/fhir/smart-app-launch/scopes-and-launch-context/#clinical-scope-syntax
Launch scopes: http://www.hl7.org/fhir/smart-app-launch/scopes-and-launch-context/#requesting-context-with-scopes